bbq/compiler/desugar.go

```diff
@@ -464,7 +464,26 @@ var panicFuncInvocationTypes = sema.InvocationExpressionTypes{
 	},
 }
 
-func (d *Desugar) desugarCondition(condition ast.Condition, isInherited bool) ast.Statement {
+func (d *Desugar) desugarCondition(condition ast.Condition, inheritedFrom *sema.InterfaceType) ast.Statement {
+
+	// If the conditions are inherited, they could be referring to their own imports.
+	// i.e: transitive dependencies of the concrete type.
+	// Therefore, add those transitive dependencies to the current compiling program.
+	// They will only be added to the final compiled program, if those are used in the code.
+	if inheritedFrom != nil {
+		elaboration, err := d.config.ElaborationResolver(inheritedFrom.Location)
+		if err != nil {
+			panic(err)
+		}
+
+		// Order doesn't matter, since imports are only added to globals in the order of usage.
```

Comment:
Does this rely on `addImport` only having "order-independent" side-effects? It looks like that might not be the case, as it appends to `d.newImports`.

Reply:
ah no, even though it appends to `d.newImports` (and eventually to import declarations of the program), when the compiler visits these import declarations, all it does is populate another map with imported symbols: https://github.com/onflow/cadence/blob/0747673c026bad007a37f70fd7893af9df2d1c1f/bbq/compiler/compiler.go#L2136

This is again order-independent.

The order is only determined by the order in which these imported symbols are used in the code. i.e: their index is set when the symbol is used in the program for the first time: https://github.com/onflow/cadence/blob/0747673c026bad007a37f70fd7893af9df2d1c1f/bbq/compiler/compiler.go#L175-L181

So the order of import declarations doesn't really matter in this case.

Reply:
Makes sense. Sorted the transitive-imports before adding them here: https://github.com/onflow/cadence/pull/3884/commits/684d094d5d628a671f4298bf0a5a54cd533d78f3